### ISSUE TITLE
[Chapter 11] SOLUTION AVAILABLE!

### ISSUE THREAD

**Issue by SuperYorio** [ageron/handson-ml2#120]
Hello guys!

Realizing that the exercises in the second half of the book has no solutions, I have decided to post my solutions to my own repo. That said, I just uploaded my solution to Ch.11's exercise at the end of the chapter. Take a look and compare our results (more details in the Notebook). I will try to organize my programs and post them periodically. Open an issue at my Repo for whatever questions or concerns you may have.

Thank you!

Here is the link to my GitHub Repo: https://github.com/SuperYorio/HandsOn_ML_Soln/blob/master/Ch_11_Soln.ipynb

**Comment by SuperYorio** [comment]
I will happily accept any feedbacks and suggestions! Of course, it would be my pleasure if Aurelien  provided any comments :)

**Comment by ageron** [comment]
Hi @SuperYorio ,
Thanks a lot for posting the solutions to the coding exercises. I've recently moved to New Zealand and had a lot of time-consuming (and stress inducing) stuff that made it difficult for me to work on the solutions until now, so your help is most welcome. 👍 
Cheers,
Aurélien

**Comment by SuperYorio** [comment]
You gave us a great book! The least I could do is just share my own results with others for referencing and comparison :)
I just feel sketchy with the results since it feels quite awkward using DNN on CIFAR10.
Overall, I also had to train my models using only 15 epochs per fit due to limited time and computation power.

Again, thank you for such a great book!

**Comment by vidyabhandary** [comment]
Hi @SuperYorio,
For step d). Try replacing Batch Normalization with SELU, and make the necessary adjustements to ensure the network self-normalizes (i.e., standardize the input features, use LeCun normal initialization, make sure the DNN contains only a sequence of dense layers, etc.).

I found that with the following code for the inputs (taken from Ageron's original notebook), the accuracy is comparable to what one gets with Batch Normalization. 

```
# Scale the inputs to mean 0 and standard deviation 1
pixel_means = X_train.mean(axis=0, keepdims=True)
pixel_stds = X_train.std(axis=0, keepdims=True)
X_train_scaled = (X_train - pixel_means) / pixel_stds
X_valid_scaled = (X_valid - pixel_means) / pixel_stds
X_test_scaled = (X_test - pixel_means) / pixel_stds

```
This change however does affect accuracy for  AlphaDropout or MCDropout. Those still hover at the same low values.

**Comment by SuperYorio** [comment]
Hello @vidyabhandary , I agree that normalizing the batch of samples is the correct and better way to do so. But since we are comparing different neural networks' relative performance, I decided to train the model on the organic dataset (the only modification is by /255.0 for the whole dataset). Thanks for sharing! I will be trying out new ways of training the model!

**Comment by ageron** [comment]
Hi there,
I just pushed the solution to the coding exercises in chapter 11.
Thanks for your patience and for coding a solution first! I'll add some thanks in the notebook as well.
Cheers,
Aurélien
